src/main/resources/assets/computercraft/lua/bios.lua

```diff
@@ -1,5 +1,48 @@
+local native_select, native_type = select, type
+
+--- Expect an argument ot have a specific type
+--
+-- @tparam int index The 1-based argument index
+-- @param value The argument's value
+-- @tparam string ... The allowed types of the argument
+-- @throws If the value is not one of the allowed types
+local function expect(index, value, ...)
+    local t = native_type(value)
+    for i = 1, native_select("#", ...) do
+        if t == native_select(i, ...) then return true end
+    end
+
+    local types = table.pack(...)
+    for i = types.n, 1, -1 do
+        if types[i] == "nil" then table.remove(types, i) end
+    end
+
+    local type_names
+    if #types <= 1 then
+        type_names = tostring(...)
+    else
+        type_names = table.concat(types, ", ", 1, #types - 1) .. " or " .. types[#types]
+    end
+
+    -- If we can determine the function name with a high level of confidence, try to include it.
+    local name
+    if native_type(debug) == "table" and native_type(debug.getinfo) == "function" then
+        local ok, info = pcall(debug.getinfo, 3, "nS")
+        if ok and info.name and #info.name ~= "" and info.what ~= "C" then name = info.name end
+    end
+
+    if name then
+        error( ("bad argument #%d to '%s' (expected %s, got %s)"):format(index, name, type_names, t), 3 )
+    else
+        error( ("bad argument #%d (expected %s, got %s)"):format(index, type_names, t), 3 )
+    end
+end
+
+-- We expose expect in the global table as APIs need to access it, but surround
+-- it with earmuffs. expect is an internal function, and should not be used by
+-- users.
+_G["~expect"] = expect
```

Comment:
I wouldn't describe one ~ as earmuffs. I understand the point but I think the description needs improving.

Perhaps unspeakable name? This is a term borrowed from C# so it kinda works.

Reply:
Ahhhh, it was originally earmuffs and then I changed it without updating the comment. Good catch!